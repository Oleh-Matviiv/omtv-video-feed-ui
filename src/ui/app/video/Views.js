import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Div = styled.div.attrs({
  className: 'video-views',
})`
  padding: 1rem;
  grid-area: views;
  text-align: right;
`;

export const formatCount = (count) => {
  const b = count / 1000000000;
  const m = count / 1000000;
  const k = count / 1000;
  if (b > 1) return `${ Math.trunc(b * 10) / 10 }B`;
  if (m > 1) return `${ Math.trunc(m * 10) / 10 }M`;
  if (k > 1) return `${ Math.trunc(k * 10) / 10 }K`;
  return `${count}`;
};

const Views = ({ count = 0 }) => (
  <Div>
    { count > 0 ? formatCount(count) + ' views' : '' }
  </Div>
);
Views.propTypes = {
  count: PropTypes.number,
};

export default Views;
