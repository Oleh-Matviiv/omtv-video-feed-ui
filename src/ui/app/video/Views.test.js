import React from 'react';
import { mount, shallow } from 'enzyme';
import 'jest-styled-components';

import Views, { Div, formatCount } from './Views';

describe('snapshots', () => {
  it('formatCount', () => {
    expect(formatCount(123)).toEqual('123');
    expect(formatCount(1123)).toEqual('1.1K');
    expect(formatCount(4231123)).toEqual('4.2M');
    expect(formatCount(9834231123)).toEqual('9.8B');
  });
  it('Div', () => {
    const wrapper = mount(
      <Div>children</Div>
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Views', () => {
    const wrapper = shallow(
      <Views count={ 2345 } />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Views - count undefined', () => {
    const wrapper = shallow(
      <Views />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
