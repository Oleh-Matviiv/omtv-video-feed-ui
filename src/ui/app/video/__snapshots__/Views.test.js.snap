// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Div 1`] = `
.c0 {
  padding: 1rem;
  grid-area: views;
  text-align: right;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "video-views",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  padding: 1rem;
  grid-area: views;
  text-align: right;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="video-views c0"
    >
      children
    </div>
  </StyledComponent>
</styled.div>
`;

exports[`snapshots Views - count undefined 1`] = `<styled.div />`;

exports[`snapshots Views 1`] = `
<styled.div>
  2.3K views
</styled.div>
`;
